### ISSUE TITLE
genfromtxt error while following doc examples

### ISSUE THREAD

**Issue by wildflower** [numpy/numpy#9653]
I'm just trying to run the examples from 
https://docs.scipy.org/doc/numpy/reference/generated/numpy.genfromtxt.html

I've tried  Anaconda 4.4 and Python 3.6.2 (64bit)
```
>>> from io import StringIO
>>> import numpy as np
>>> s = StringIO("1,1.3,abcde")
>>> data = np.genfromtxt(s, dtype=[('myint','i8'),('myfloat','f8'),('mystring','S5')], delimiter=",")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Program Files\Python36\lib\site-packages\numpy\lib\npyio.py", line 1580, in genfromtxt
    first_values = split_line(first_line)
  File "C:\Program Files\Python36\lib\site-packages\numpy\lib\_iotools.py", line 220, in _delimited_splitter
    line = line.split(self.comments)[0]
TypeError: must be str or None, not bytes
```
the above error is on Windows

I get a similar error on Ubuntu python 3.5.2
```
pip3 install numpy
Collecting numpy
  Downloading numpy-1.13.1-cp35-cp35m-manylinux1_x86_64.whl

Python 3.5.2 (default, Nov 17 2016, 17:05:23) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> from io import StringIO
>>> s = StringIO("1,1.3,abcde")
>>> data = np.genfromtxt(s, dtype=[('myint','i8'),('myfloat','f8'),('mystring','S5')], delimiter=",")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/haydn/.local/lib/python3.5/site-packages/numpy/lib/npyio.py", line 1580, in genfromtxt
    first_values = split_line(first_line)
  File "/home/haydn/.local/lib/python3.5/site-packages/numpy/lib/_iotools.py", line 220, in _delimited_splitter
    line = line.split(self.comments)[0]
TypeError: Can't convert 'bytes' object to str implicitly
>>> data = np.genfromtxt(s, dtype=[('myint','i8'),('myfloat','f8'),('mystring','str')], delimiter=",")
__main__:1: UserWarning: genfromtxt: Empty input file: "<_io.StringIO object at 0x7feb518cff78>"
>>> s = StringIO("1,1.3,abcde")
>>> data = np.genfromtxt(s, dtype=[('myint','i8'),('myfloat','f8'),('mystring','str')], delimiter=",")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/haydn/.local/lib/python3.5/site-packages/numpy/lib/npyio.py", line 1580, in genfromtxt
    first_values = split_line(first_line)
  File "/home/haydn/.local/lib/python3.5/site-packages/numpy/lib/_iotools.py", line 220, in _delimited_splitter
    line = line.split(self.comments)[0]
TypeError: Can't convert 'bytes' object to str implicitly
```

**Comment by adeak** [comment]
This was a bug (or a mistake in the docs, depending on perspective), probably related to https://github.com/numpy/numpy/issues/3184. Upgrading to numpy 1.14 should fix it for python 3.

**Comment by eric-wieser** [comment]
Yep, this is indeed fixed in 1.14
